OpenRA.Mods.TS/Traits/Render/WithVoxelWaterBody.cs

```diff
@@ -0,0 +1,73 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.TS.Traits
+{
+	public class WithVoxelWaterBodyInfo : ITraitInfo, IQuantizeBodyOrientationInfo, IRenderActorPreviewVoxelsInfo, Requires<RenderVoxelsInfo>
+	{
+		public readonly string WaterSequence = "water";
+		public readonly string LandSequence = "idle";
+
+		public object Create(ActorInitializer init) { return new WithVoxelWaterBody(init.Self, this); }
+
+		public IEnumerable<VoxelAnimation> RenderPreviewVoxels(ActorPreviewInitializer init, RenderVoxelsInfo rv, string image, WRot orientation, int facings, PaletteReference p)
+		{
+			var body = init.Actor.Traits.Get<BodyOrientationInfo>();
+			var voxel = VoxelProvider.GetVoxel(image, LandSequence);
+			yield return new VoxelAnimation(voxel, () => WVec.Zero,
+				() => new[] { body.QuantizeOrientation(orientation, facings) },
+				() => false, () => 0);
+		}
+
+		public int QuantizedBodyFacings(ActorInfo ai, SequenceProvider sequenceProvider, string race) { return 0; }
+	}
+
+	public class WithVoxelWaterBody : IAutoSelectionSize
+	{
+		bool Water { get { return self.World.Map.GetTerrainInfo(self.Location).IsWater; } }
```

Comment:
Our naming conventions for bools are <preposition><noun>, so I suggest calling this `OverWater`.  It should also live below the readonly fields.